### DISCUSSION TITLE
iOS 倉輸入法 可使用 電腦 Rime 洋蔥雙拼注音 & 手機同文輸入法洋蔥雙拼注音

### DISCUSSION THREAD

**Discussion by sjackwu** [oniondelta/Onion_Rime_Files#31] [Ideas]
FYI，回報一下。這幾天看到這個 iOS 上的 RIME 輸入法試用之後，電腦版跟手機版的洋蔥雙拼注音都能使用。

電腦版：lua （打今天日期時間）& opencc （輸入表情符號）都能使用
手機版：能使用 opencc （輸入表情符號）。還沒有嘗試把手機版的鍵盤檔案改成洋蔥版的...

[在 App Store 上的「仓输入法」](https://apps.apple.com/tw/app/%E4%BB%93%E8%BE%93%E5%85%A5%E6%B3%95/id6446617683)

[imfuxiao/Hamster: librime for iOS App](https://github.com/imfuxiao/Hamster)

這張圖裡我用倉輸入法的UI設定一些自訂鍵盤，還在嘗試中
![image](https://github.com/oniondelta/Onion_Rime_Files/assets/17001440/ae98a367-3cd3-4125-ab08-d7e54785a7cd)

**Comment by oniondelta** [comment]
感謝反餽！😄
鍵盤檔應該與 android 不相通？！
android 的鍵盤檔不同版本也存在差異，有些寫法無法通用，或在新版有 Bug XD

**Comment by sjackwu** [comment]
鍵盤檔...有點看不懂，就沒研究要怎麼要怎麼放進去 😅
但倉輸入法本身的設定UI裡面可以修改鍵盤，就弄出一個鍵盤來，但可能有點玩上癮了↓😄
FYI [「仓」输入法简介 · HonKit](https://ihsiao.com/apps/hamster/docs/)
![image](https://github.com/user-attachments/assets/6ea03f93-0ac2-4612-8e9b-3413ffb0d8e4)

**Comment by oniondelta** [comment]
看了連結文件，兩個鍵盤檔不通用！

iOS 上記得還有 [iRime](https://apps.apple.com/tw/app/irime%E8%BE%93%E5%85%A5%E6%B3%95-%E5%B0%8F%E9%B9%A4%E5%8F%8C%E6%8B%BC%E4%BA%94%E7%AC%94%E9%83%91%E7%A0%81%E8%BE%93%E5%85%A5%E6%B3%95/id1142623977)，[其GitHub](https://github.com/jimmy54/iRime)，一開始還不能改鍵盤檔，後來可以！
手邊沒 iOS 設備，故也沒設 iRime 其鍵盤檔！
感覺新的 Hamster 仓输入法很不錯！

Hamster 仓输入法應該蠻絲滑的？！
android 這邊 trime 早期也算絲滑，現在會卡頓！
設定檔怎調都一樣，且 bug 多。
當初 trime 也玩上癮！但現在會卡頓，就先不推 trime！

**Comment by sjackwu** [comment]
現在倉輸入法能直接從UI上調整鍵盤，比較適合我 😀
iRime, 以洋蔥注音雙拼來說，不曉得為什麼，新版的（洋蔥注音）會重新部署後打不出字，逼我只能回到過去自己存檔（確定能用）的舊版。倉輸入法能直接用最新版本的電腦版&手機版的洋蔥注音，

目前倉輸入法在 iphone 13 & iphone 6s 上，搭配洋蔥注音雙拼，都沒看到卡頓的狀況。

我自己在電腦上習慣套用 不開頭簡拼custom（遇到要翻頁找字的時候會瘋掉），但在手機上，試著不要套用，覺得比較能體會那個速度感（因為候選字可以不用只有一排，而是能一次看到三四排，一眼就能看到一頁二三十個，找起來比較快）。現在問題是已經習慣每個字都打兩碼的手感了，一下子要換到只打開頭簡拼有點不習慣😅

**Comment by oniondelta** [comment]
也遇過「洋蔥注音雙拼」在小狼毫重新部署後打不出字，推測原因為 [引用和部屬先後關係發生問題](https://github.com/oniondelta/Onion_Rime_Files/wiki/%E3%80%8E-%E6%B3%A8%E9%9F%B3-%E6%B4%8B%E8%94%A5-%E9%9B%99%E6%8B%BC-%E7%89%88-%E3%80%8F%E6%96%B9%E6%A1%88%E8%AA%AA%E6%98%8E#%E7%84%A1%E6%B3%95%E5%AE%8C%E6%88%90%E9%83%A8%E7%BD%B2%E6%99%82)，不知 iRime 是否為此因素，或新版加了一些功能所產生的問題？！

關於開頭簡不簡拼，在設定時也猶豫且測試許久，更新有多次調整正則（簡不簡拼都有調整）：

- 「不簡拼」：新更新的正則調整有點皺眉！因應新版 librime 核心功設定 `enable_completion: true` 會補詞（如：只輸入「好」該字，「好的」等詞庫有的詞會出現在選單中），設置 `enable_completion: false` 可把 librime 該新功能關掉，目前設定為關掉，但會使的輸入首碼成為空碼，設置為 true 則不會，但選字又跑出補詞XD， [該問題相關討論](https://github.com/rime/librime/issues/905)。

- 「簡拼」找字較麻煩，電腦上：1. 建議不要一次打太長句。 2.善用「 Tab 鍵（首碼）」從開頭開始去選字（搭配方向鍵）。 3. 善用「五聲調」去限定，[「聲調限定」相關說明](https://github.com/oniondelta/Onion_Rime_Files/wiki/%E3%80%8E-%E6%B3%A8%E9%9F%B3-%E6%B4%8B%E8%94%A5-%E9%9B%99%E6%8B%BC-%E7%89%88-%E3%80%8F%E6%96%B9%E6%A1%88%E8%AA%AA%E6%98%8E#-%E6%B4%8B%E8%94%A5%E9%9B%99%E6%8B%BC%E6%B3%A8%E9%9F%B3%E7%B0%A1%E5%8C%96%E8%BC%B8%E5%85%A5)。承上，手機鍵盤建議要設置「 Tab 鍵（首碼）」和「五聲調」去限定。

- 「不簡拼」和「簡拼」的差異就像電腦上的「新注音」和手機上可首碼輸入的「注音」。如同「哈密瓜🍈」XD 

- 如要手機「不簡拼」，把電腦版的「bopomo_onion_double.custom.yaml」該檔更名為「bpmfmobiledouble.custom.yaml」置入並重新部屬即可。
